### PR TITLE
Don't consume async when checking for async func decl

### PR DESCRIPTION
**Pull request by existentialism**
As reported in https://github.com/babel/babel/issues/5366.

### CODE REVIEW

File: src/parser/statement.js [hunk 1]
```diff
@@ -835,6 +835,12 @@ pp.parseExport = function (node) {
     let needsSemi = false;
     if (this.eat(tt._function)) {
       expr = this.parseFunction(expr, true, false, false, true);
+    } else if (
```
Comment (existentialism) [inline src/parser/statement.js]: @danez not sure if there's a cleaner way to do this check w/o consuming?

**Comment by danez** [comment]
Isn't this the exact same thing as the just reverted commit? This would reintroduce the problem. The problem is that babel can't handle the new AST.

**Comment by danez** [comment]
Oh I see this is about arrow functions. Before we merge this we need to fix babel first to understand the AST.

**Comment by hzoo** [comment]
If necessary we can backport to 6.x

**Pull request merged**